pkg/subscriber/helmrepo/helm_subscriber_item.go

```diff
@@ -127,10 +127,37 @@ func (hrsi *SubscriberItem) doSubscription() {
 	hrNames := getHelmReleaseNames(indexFile, hrsi.Subscription)
 
 	for _, hrName := range hrNames {
-		if hash != hrsi.hash || !hrsi.success ||
-			!isHelmReleaseExists(hrsi.synchronizer.GetLocalClient(), hrsi.Subscription.Namespace, hrName) ||
-			!isHelmReleaseStatusUpdated(hrsi.synchronizer.GetLocalClient(), types.NamespacedName{Name: hrsi.Subscription.Name,
-				Namespace: hrsi.Subscription.Namespace}, hrsi.Subscription.Namespace, hrName) {
+		isHashDiff := hash != hrsi.hash
+		isUnsuccessful := !hrsi.success
+		existsHelmRelease := false
+		populatedHelmReleaseStatus := false
+
+		existsHelmRelease, err = isHelmReleaseExists(hrsi.synchronizer.GetLocalClient(), hrsi.Subscription.Namespace, hrName)
+		if err != nil {
+			klog.Error("Failed to determine if HelmRelease exists: ", err)
+
+			hrsi.success = false
+
+			return
+		}
+
+		if existsHelmRelease {
+			populatedHelmReleaseStatus, err = isHelmReleaseStatusPopulated(hrsi.synchronizer.GetLocalClient(),
+				types.NamespacedName{Name: hrsi.Subscription.Name,
+					Namespace: hrsi.Subscription.Namespace}, hrsi.Subscription.Namespace, hrName)
+			if err != nil {
+				klog.Error("Failed to determine if HelmRelease status is populated: ", err)
+
+				hrsi.success = false
+
+				return
+			}
+		}
+
+		if isHashDiff || isUnsuccessful || !existsHelmRelease || !populatedHelmReleaseStatus {
+			klog.Infof("Processing Helm Subscription... isHashDiff=%v isUnsuccessful=%v existsHelmRelease=%v populatedHelmReleaseStatus=%v",
```

Comment:
I like this logging.

Reply:
Trying to follow our new logging guideline :)